java/ql/test/library-tests/pathsanitizer/Test.java

```diff
@@ -604,4 +604,214 @@ public void fileConstructorSanitizer() throws Exception {
+            }
+        }
+        {
+            String source = (String) source();
+            // exclude '.', '/', '\'
+            if (source.matches("[^0-9./\\\\a-f]+")) {
+                sink(source); // Safe
+            } else {
+                sink(source); // $ hasTaintFlow
+            }
+        }
+        {
+            String source = (String) source();
+            // '/' is not excluded
+            if (source.matches("[^0-9.\\\\a-f]+")) {
+                sink(source); // $ hasTaintFlow
+            } else {
+                sink(source); // $ hasTaintFlow
+            }
+        }
+        // branch = false
+        {
+            String source = (String) source();
+            if (source.matches("[\\./\\\\]+")) {
+                sink(source); // $ hasTaintFlow
+            } else {
+                sink(source); // $ Safe
+            }
+        }
+        {
+            String source = (String) source();
+            // not a complete sanitizer since it doesn't protect against absolute path injection
+            if (source.matches("[\\.]+")) {
+                sink(source); // $ hasTaintFlow
+            } else {
+                sink(source); // $ hasTaintFlow
+            }
+        }
+        // validation method
+        {
+            String source = (String) source();
+            directoryCharsValidation(source);
+            sink(source); // Safe
+        }
+
+        // ReplaceDirectoryCharactersSanitizer
+        // Removes ".." sequences and path separators from the payload
+        // single `replaceAll` call
+        {
+            String source = (String) source();
+            source = source.replaceAll("\\.\\.|[/\\\\]", "");
+            sink(source); // Safe
+        }
+        {
+            String source = (String) source();
+            source = source.replaceAll("\\.|[/\\\\]", "");
+            sink(source); // Safe
+        }
+        {
+            String source = (String) source();
+            source = source.replaceAll("[.][.]|[/\\\\]", "");
+            sink(source); // Safe
+        }
+        {
+            String source = (String) source();
+            source = source.replaceAll(".|[/\\\\]", "");
+            sink(source); // $ hasTaintFlow
+        }
+        {
+            String source = (String) source();
+            source = source.replaceAll("[\\./\\\\]", "");
+            sink(source); // Safe
+        }
+        {
+            String source = (String) source();
+            source = source.replaceAll("[\\.\\\\]", "");
+            sink(source); // $ hasTaintFlow
+        }
+        {
+            String source = (String) source();
+            source = source.replaceAll("[^\\.\\\\/]", "");
+            sink(source); // $ hasTaintFlow
+        }
+        {
+            String source = (String) source();
+            // Bypassable with ".../...//"
+            source = source.replaceAll("\\.\\./", "");
+            sink(source); // $ hasTaintFlow
+        }
```

Comment:
Add a test of the form "A|B|C", like this (which I checked works):
```suggestion
        }
        {
            String source = (String) source();
            source = source.replaceAll("\\.|/|\\\\", "");
            sink(source); // Safe
        }
```

Reply:
Added in https://github.com/github/codeql/pull/18646/commits/2f6696e8a8816afd2ecb7236529ea3023978d3d1.